src/Navigation/Drawer/Drawer.tsx

```diff
@@ -9,42 +9,30 @@ const Drawer = ({
   position = 'right',
   ...defaultProps
 }: Props) => {
-  const [isDisplay, setIsDisplay] = React.useState<boolean>(false);
-
-  const handleAnimationStart = () => {
-    setIsDisplay(true);
-  };
-
-  const handleAnimationEnd = () => {
-    setIsDisplay(false);
-  };
-
+  const [firstRenderDone, setFirstRenderDone] = React.useState(false);
   React.useEffect(
-    function startAnimationOnOpen() {
-      if (isOpen && !isDisplay) {
-        handleAnimationStart();
+    function markFirstRenderAsDone() {
```

Comment:
So this flag is only required for the animation? Hmm I guess it's a hacky way and someone might not understand the purpose while going through the component easily?
Maybe we should add a comment with a TO-DO to fix the way?